src/lib/DeployUtil.ts

```diff
@@ -1241,7 +1241,14 @@ export const validateDeploy = (deploy: Deploy): Result<Deploy, string> => {
                   got: ${deploy.hash}.`);
   }
 
-  // TODO: Verify included signatures.
+  deploy.approvals.forEach(({ signer, signature }) => {
+    const pk = CLPublicKey.fromHex(signer);
+    const signatureRaw = decodeBase16(signature.slice(2));
+    const isValid = validateSignature(deploy.hash, signatureRaw, pk);
+    if (!isValid) throw Error('Invalid signature');
```

Comment:
This seems inconsistent with the overall function. It should return an Err() instead of throwing an Error() no ?

Reply:
this is already resolved